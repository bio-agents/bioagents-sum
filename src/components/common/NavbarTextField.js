import React from 'react'
import { FormControl, FormGroup } from 'react-bootstrap'

const NavbarTextField = ({ placeholder, input }) => (
  <FormGroup>
    <FormControl type='text' placeholder={placeholder} {...input} />
  </FormGroup>
)

export default NavbarTextField
