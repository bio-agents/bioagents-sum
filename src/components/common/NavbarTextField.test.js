import React from 'react'
import renderer from 'react-test-renderer'
import NavbarTextField from './NavbarTextField'

it('Renders NavbarTextField with placeholder correctly', () => {
  const tree = renderer
    .create(<NavbarTextField placeholder='Write something' />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Renders NavbarTextField without placeholder correctly', () => {
  const tree = renderer
    .create(<NavbarTextField />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
