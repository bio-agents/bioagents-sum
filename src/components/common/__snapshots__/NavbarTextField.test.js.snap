exports[`test Renders NavbarTextField with placeholder correctly 1`] = `
<div
  className="form-group">
  <input
    className="form-control"
    id={undefined}
    placeholder="Write something"
    type="text" />
</div>
`;

exports[`test Renders NavbarTextField without placeholder correctly 1`] = `
<div
  className="form-group">
  <input
    className="form-control"
    id={undefined}
    placeholder={undefined}
    type="text" />
</div>
`;
